flow-nodes/api-builder-plugin-petstore/test.js

```diff
@@ -0,0 +1,74 @@
+const { expect } = require('chai');
+const { MockRuntime } = require('@axway/api-builder-sdk');
+
+const getPlugin = require('./index');
+
+function getLogger () {
+	const logger = {
+		debug: () => {},
+		trace: () => {},
+		info: () => {},
+		warn: () => {},
+		error: () => {},
+		fatal: () => {}
+	};
+	logger.scope = () => logger;
+	return logger;
+}
+
+describe('flow-node petstore', () => {
+	let runtime;
+	before(async () => {
+		const plugin = await getPlugin({}, {
+			logger: getLogger()
+		});
+		runtime = new MockRuntime(plugin);
+	});
+
+	describe('#constructor', () => {
+		it('should define flow-nodes', () => {
+			expect(runtime).to.exist;
+			// Ensure there's a flow-node and schemas created for each OAS document
+			expect(Object.keys(runtime.plugin.flownodes)).to.have.length(1);
+			expect(runtime.plugin.schema).to.have.length(6);
+			const flownode = runtime.getFlowNode('petstore');
+			expect(flownode).to.be.a('object');
+
+			// Ensure the flow-node matches the OAS document
+			expect(flownode.name).to.equal('Swagger Petstore');
+			expect(flownode.description).to.equal('This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.');
+			expect(flownode.icon).to.be.a('string');
+			expect(Object.keys(flownode.methods)).to.deep.equal([
```

Comment:
I know we want to keep things simplified here and checking the vary least all methods are there is good. However, I think it is important to show that the flow-node methods match the OAS document - params, outputs, etc. Maybe we could add a  single method sample to showcase this. People would want to cover these in their real plugins.